OpenAPI/kiota/tutorials/dotnet-dependencyinjection.md

```diff
@@ -0,0 +1,288 @@
+    }
+}
+```
+
+## Register the API client
+
+Open the _Program.cs_ file and add the following code above the `var app = builder.Build();` line.
+
+```csharp
+using GitHub;
+...
+
+// Add Kiota handlers to the DI container
+builder.Services.AddKiotaHandlers();
+
+// Register the factory for the GitHub client
+builder.Services.AddHttpClient<GitHubClientFactory>((sp, client) => {
+    client.BaseAddress = new Uri("https://api.github.com");
+    client.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
+}).AttachKiotaHandlers();
+
+// Register the GitHub client
+builder.Services.AddTransient(sp=> sp.GetRequiredService<GitHubClientFactory>().GetClient());
+```
+
+## Create an endpoint
+
+We will now create an endpoint that uses the GitHub client from dependency injection to get the latest release of `dotnet/runtime`.
+
+Open the _Program.cs_ file and add the following code above the `app.Run();` line.
+
+```csharp
+app.MapGet("/dotnet/releases", async (GitHub.ApiClient.GitHubClient client, CancellationToken cancellationToken) =>
+{
+    var releases = await client.Repos["dotnet"]["runtime"].Releases.Latest.GetAsync(cancellationToken: cancellationToken);
+    return releases;
+})
+.WithName("GetDotnetReleases")
+.WithOpenApi();
+```
+
+## Run the application
+
+Run the following command in your project directory to start the application.
+
+```dotnetcli
+dotnet run
+```
+
+You can now open a browser and navigate to `http://localhost:{port}/dotnet/releases` to see the latest release of `dotnet/runtime`. Alternatively, you can also go to `http://localhost:{port}/swagger` to see the Swagger UI with this new endpoint.
+
+## Authentication
+
+The `GitHubClientFactory` class is responsible for creating the client, and it is also responsible for creating the `IAuthenticationProvider`. In this sample, we are using the `AnonymousAuthenticationProvider`, which is a simple provider that does not add any authentication headers. More details on authentication with Kiota can be found in the [Kiota authentication documentation](/openapi/kiota/authentication?wt.mc_id=SEC-MVP-5004985).
+
+What to remember is that the `GitHubClientFactory` is used through dependency injection, which means you can have other types injected by the constructor that are needed for the authentication provider.
+
+In [this example](https://github.com/svrooij/KiotaWithDependencyInjection/blob/d1ee972e8ba06c4f9707b78dfa981f5f57fbf7e6/KiotaWithDependencyInjection/Spotify/SpotifyFactory.cs) there is an `IAccessTokenProvider` injected that is then [used](https://github.com/svrooij/KiotaWithDependencyInjection/blob/d1ee972e8ba06c4f9707b78dfa981f5f57fbf7e6/KiotaWithDependencyInjection/Spotify/SpotifyFactory.cs#L36) for the [BaseBearerTokenAuthenticationProvider](/openapi/kiota/authentication?tabs=csharp&wt.mc_id=SEC-MVP-5004985#base-bearer-token-authentication-provider) that is available in Kiota.
```

Comment:
links to "3rd party repositories" will lead to scrutiny. I suggest removing this section

Reply:
No sample better than me creating a pr for the samples repo?

Reply:
we could alternatively do that if you feel like adding more samples to the kiota-samples repo

Reply:
Might create the sample in the sample repo and then create a new pr to update this one.